repo/blob/azure/patch.go

```diff
@@ -0,0 +1,34 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/kopia/kopia/internal/clock"
+	"github.com/kopia/kopia/repo/blob"
+	"github.com/kopia/kopia/repo/blob/retrying"
+	"github.com/pkg/errors"
+)
+
+// NewWithClient creates new Azure backend storage with the specified client
+func NewWithClient(ctx context.Context, opt *Options, client *azblob.Client) (blob.Storage, error) {
+	raw := &azStorage{
+		Options:   *opt,
+		container: opt.Container,
```

Comment:
I see in the original code (in func `azure.New`), bucket (as `azblob.ContainerClient`) field is set, here we set the `container` as string instead, is it ok to call the same `retrying.NewWrapper `later?

```
	bucket := service.NewContainerClient(opt.Container)

	raw := &azStorage{
		Options: *opt,
		bucket:  bucket,
		service: service,
	}
```